receipt_label/receipt_label/utils/text_reconstruction.py

```
@@ -0,0 +1,248 @@
+        Reconstruct receipt text from ReceiptLine entities using _format_receipt_lines approach.
+        Groups visually contiguous lines based on Y coordinates.
+
+        Returns:
+            Tuple of (formatted_text, list_of_text_groups)
+        """
+        if not lines:
+            return "", []
+
+        # Sort lines by Y position (top to bottom) then by line_id
+        sorted_lines = sorted(
+            lines, key=lambda l: (-l.calculate_centroid()[1], l.line_id)
```

Ruff (0.12.2)

Ambiguous variable name: `l`

(E741)